### ISSUE TITLE
Can't delete image that is in a DCIM subfolder

### ISSUE THREAD

**Issue by KryptoSynth** [deckerst/aves#804]
**Describe the bug**
Giving storage access for an image located inside a DCIM subfolder and grants access to "DCIM", resulting in a failed message upon pressing delete. 

But doing the same for an image directly inside the DCIM folder grants access to a "dcim" folder and allows for successful deletion of media both in the DCIM folder and its subfolders.

The "DCIM" storage access entry (in Aves settings) doesn't seem correlate with the actual folder on the device (but the "dcim" entry does.)

**To Reproduce**
1. Have an image in DCIM/*subfolder*/*image.jpg*
2. Press delete button.
3. Give storage access to "DCIM" folder.
4. See error message.

**Subsequently**
1. Have an image in DCIM/*image.jpg*
2. Press delete and grant access to "dcim" folder.
3. No error and successful deletion.

**Expected behavior**
Giving storage access for an image in a DCIM subfolder should grant access correctly.

**Screenshots**
![Screenshot_2023_1029_231312](https://github.com/deckerst/aves/assets/109325672/3449d7f0-0ee2-4b99-acbf-f38b32a36ac3)

**System information and logs:**
Package: deckers.thibault.aves
Installer: com.android.vending
Aves version: 1.9.7-play, build 108
Flutter: stable 3.13.7
Android version: 13, API 33
Android build: TP1A.220624.014 release-keys
Device: vivo I2011
Geocoder: ready
Mobile services: ready
System locales: en_US
Storage volumes: /storage/emulated/0/, /storage/C0B5-9BF6/
Storage grants: /storage/emulated/0/Movies/, /storage/emulated/0/Pictures/, /storage/emulated/0/Documents/, /storage/emulated/0/DCIM/, /storage/emulated/0/dcim/
Error reporting: true

[aves-logs-20231029_232523.txt](https://github.com/deckerst/aves/files/13198813/aves-logs-20231029_232523.txt)

**Additional context**
The above image was in DCIM/Screenshots.

**Comment by deckerst** [comment]
I'll take a closer look at this later, but for now i have a quick question: i see in your grants that there is both "/storage/emulated/0/DCIM/" and "/storage/emulated/0/dcim/". How is that possible? Was the folder name at some point "DCIM", you gave permission for it, then deleted it and recreated it as "dcim"?

**Comment by KryptoSynth** [comment]
"DCIM" entry was created when I granted permission for an image located inside "/DCIM/Screenshots/". "dcim" was created when I granted permission for an image located directly inside "/DCIM/". I never renamed any folder. Aves apparently recognizes the folder differently based on where the image is.

**Comment by deckerst** [comment]
I see. So the first weird thing is that the app should never have asked you for another permission when editing the image directly inside DCIM, as it already had the permission.

I've dealt with some file path casing issues before and it's tricky. I'll need to try and reproduce your situation on my device.

**Comment by deckerst** [comment]
I've tried to reproduce the issue on my device, to no avail (and the logs did not yield anything useful).

Actually, I don't understand why Aves would ask/get permission as "dcim" if the directory name is indeed `DCIM` in uppercase. All the rest is probably a consequence of this weirdness.

If you clear the grants for `DCIM` and `dcim` via Settings > Privacy > Storage access, then can you reproduce the bug at will by following the same sequence of actions you indicated in your reproduction scenario above?

Also, do you use a custom ROM?

**Comment by KryptoSynth** [comment]
No custom ROM, just FuntouchOS (vivo/iqoo ROM).

I have discovered a new behaviour with user-created directories in DCIM. It's in the linked video.

https://crepe.moe/kESinNul

In the video, the "placeholder" folder was created by me, and "Screenshots" was created by the OS.

And yes, it is completely reproducable.

**Comment by deckerst** [comment]
Thanks for digging further.

The first mystery to me is why the system file picker shows the directory as "dcim" while mixplorer shows it as "DCIM".

Android is technically case sensitive, but normally everything is made to present it to the user as case insensitive. What I mean is that it's possible for the system to have both "dcim" and "DCIM" folders side by side but typical file managers won't let you create this situation.

My guess is that on your system you indeed have both. Different apps will show the combined content from both in a single directory called either "DCIM" or "dcim", depending on the API used and how paths are resolved, as if there was a single directory.

This is a mess. I'm not even sure i can recreate this situation without FuntouchOS.

**Comment by RenLabz** [comment]
"All File Access" would make the app redundant and useless. Drill down the issue and solving it would be the best way, and why we users take the time to post and report the issue. 

What might help is different people reporting it. Or a work around. 

It looks like if you save or move all images out of DCIM the problem should not apply. I dont know if the camera app I use has the ability to store taken photos into another folder, but I would do that if possible.

**Comment by deckerst** [comment]
You're missing the point: that app is a fundamental system app. If your device doesn't have it, it's non standard.

**Comment by aloxe** [comment]
I experienced the same thing here with Aves 1.11.13 on Android 13.

As it looked like a difficulty with storage access, I checked the storage in my settings, I saw that there were two SD card listed when I only have one. The "empty" one could be removed so I removed it. After that, even the camera could not save photos but after a reboot, it all got together and the camera as well as deleting with Aves.

I am not sure this is the same issue that everybody here has but I am sharing this experience in case that can help.

Otherwise checking the settings of Aves and unticking **Privacy** > **Use recycle bin** may also help.

**Comment by deckerst** [comment]
@aloxe Thanks for sharing your experience.

> there were two SD card listed when I only have one. The "empty" one could be removed so I removed it.

Just to be clear, you mean there were 2 SD card listed in the Android system settings? If so, was it a phantom SD card that you inserted before and had removed since then?

**Comment by aloxe** [comment]
It was a phantom card listed, but the incident was possibly due to the opening of the phone and the removal of the card earlier on. Aves took time to find the photos and resync them again. I think the phantom card is a result of this hardware hiccup.

**Comment by tteskac** [comment]
I have the same issue, this is really annoying because if you have "Use recycle bin" enabled, then you might not even notice this problem. I figured out because after a while (now I can reproduce it every time by simply killing the app and opening again) the deleted images pop up again in the main gallery. Tried to delete it many times, also from recycle bin and they alwas come back. Now when I disabled recycle bin and when I delete them it also shows just "Failed" toast, like on the screenshot of OP.

In my case "DCIM" folder is uppercase on both locations. Tried removing and granting the permission multiple ways, no success. So not sure what is going on. I use stock OS, Xiaomi HyperOS 1.0.9.0, Xiaomi 13T Pro, Android 14.

**Comment by tteskac** [comment]
Okay, I might be onto something.

Under the "Storage access" list, the "DCIM" is uppercase in the path, using file managers also uppercase. Both look correct.

But if look at the image details that cant be deleted, it shows the Path with lowercase "dcim", while other parts in the path look correct, e.g. "Camera" with uppercase C. So it's not that the whole path "got lowercased" somehow but only "dcim" part.

I have a set of images with such lowercase "dcim" which cant be deleted, **but most of other images have "DCIM" and can be deleted just normally.** So aparently it's actually only subset of files with such issue. See screenshots of two cases. **Both taken with standard camera app**, not sure what I did differently...

![image](https://github.com/user-attachments/assets/38b4a3bc-fe56-4c40-8c77-77bfd11aa25e)
![image](https://github.com/user-attachments/assets/a4823b65-e1c6-48f6-b93b-6d7b2370f4f3)

**Comment by deckerst** [comment]
Yes, I agree it's confusing. My understanding is the following. Underlying Android is a case sensitive OS that allows both "DCIM" and "dcim" to coexist, but everything is done in Android to present to the user the file storage as being case insensitive, so the user cannot create a folder or file with the same name only differing in case.

In your example, I suppose the app that created the files registered them in the Media Store with different case for "DCIM". Not sure whether it's only differing in the Media Store, or whether they are actually in 2 different folders. Anyway now the user only sees one folder "DCIM" in their file storage, merging both cases.

I would need to try different workarounds. It's easy to get wrong, as I use multiple approaches to access the file storage, depending on location.

**Comment by tteskac** [comment]
Many thanks for the reply. You are right. Seems like its not the fault of your app regarding the path representation. I actually see it in "Files by Google" app like that as well when I select info about the file. Althought **I was able to delete it there and seems like that's the workaround for such files for now** (probably with other file managers and stock gallery would work as well).
